App/automotive/src/main/java/com/amos/infotaimos/model/NavigationService.kt

```diff
@@ -82,6 +85,29 @@ object NavigationService {
         Log.d(TAG, "Updated Navigation LiveData")
     }
 
+    fun speechAnnouncement(mediaPlayer: MediaPlayer, delay: Long) {
+
+        Log.d(TAG, "Requested navigation announcement in $delay ms")
+
+        //announcementTask?.cancel()
+        if (announcementTask == null) {
+            announcementTask = Timer().schedule(delay) {
+                Log.d(TAG, "Navigation announcement timer fired")
+                mediaPlayer.start()
+
+                Thread.sleep(3000)
```

Comment:
sleep ist eigentlich immer ungeil, lieber mit Timer Funktionen machen

Reply:
hatte ich sogar mal im commit 3 "implement speech announcement with delay".
Da der code aber schon in einem Timer ausgeführt wird, dachte ich dass des so ausreichen sollte, sonst starte ich einen Timer.schedule() im Timer.schedule()